src/snowflake/snowpark/mock/_plan.py

```diff
@@ -605,11 +605,18 @@ def cleanup_imports():
             # And these code would look like:
             # res=input.apply(...)
             # res.set_sf_type(ColumnType(exp.datatype, exp.nullable))  # fixes the drift and removes NaT
+
+            strict = analyzer.session.udf.get_strictness(udf_name)
```

Comment:
I'm wondering if it would be better to keep the strict information to the udf itself (e.g, like Mocked SP, we create a MockedUserDefinedFunction with the strict info) instead of maintaining a global map to track these information to make the code structure more consistent.